### ISSUE TITLE
Bug: log(0) in module_mp_nssl_2mom.F

### ISSUE THREAD

**Issue by ccarouge** [wrf-model/WRF#1207]
I've stumbled upon this while running with debugging options to solve a different issue. I have absolutely no idea about microphysics so I'm absolutely unable to solve it.

This is with WRF v4.2.

In the file module_mp_nssl_2mom.F, from line 1016:
```
   DO i = 0,nqiacrratio
        ratio = float(i)*dqiacrratio
        x = gamxinf( 1.+alp, ratio )
```
So `ratio=0` at the first iteration. 
But then in gamxinf function (skipping some lines for clarity):
```
        real function GAMXINF(A1,X1)
        x = x1
        XAM=-X+A*DLOG(X)
```
So the code does `DLOG(0)` which obviously is problematic.

**Comment by davegill** [comment]
@ccarouge @weiwangncar @dudhia 
Claire,
Let's see what the physics team says.
Thanks for the detective work that I know was required to make this as simple as possible for us.

**Comment by weiwangncar** [comment]
@MicroTed Ted, I wonder if you can provide some input to this user's problem? Thanks.

**Comment by MicroTed** [comment]
@weiwangncar Yes, I also found this issue and have updated code that fixes it. The problem was that I called the incomplete function for a complete function, so I added a check within the function for the case of x1=0. An error shows up if floating point exceptions are checked. I need to update a couple things, too, actually, but hadn't gotten to submitting it yet. I could submit a fix just for this if you want a quick update.
